File: boj/mtak/11723.py
opcnt = int(input())
s = 0
def _add(x):
    global s
    s|=(1<<x)
def _remove(x):
    global s
    s&=~(1<<x)
def _check(x):
    print(int(s & x))
def _toggle(x):
    global s
    s^=(1<<x)
def _all():
    global s
    s = (1<<21) - 1
def _empty():
    global s
    s = 0
tool = {"add":_add, "remove":_remove, "check":_check, "toggle":_toggle, "all":_all, "empty":_empty}

for _ in range(opcnt):
    cmd = input().split()
    if len(cmd) == 2:
        tool[cmd[0]](int(cmd[1]))
    else:
        tool[cmd[0]]()

File: boj/wmo/11723.c
#include <stdio.h>
#include <string.h>

char str[7];

int main() {

	int flag;
	int n;
	int num;

	flag = 0;
	scanf("%d", &n);
	while (n--) {
		scanf("%s", str);
		if (!strcmp(str, "add")) {
			scanf("%d", &num);
			flag |= (1 << num);
		}
		else if (!strcmp(str, "check")) {
			scanf("%d", &num);
			if (flag & (1 << num))
				printf("1\n");
			else
				printf("0\n");
		}
		else if (!strcmp(str, "toggle")) {
			scanf("%d", &num);
			flag ^= (1 << num);
		}
		else if (!strcmp(str, "remove")) {
			scanf("%d", &num);
			flag &= ~(1 << num);
		}
		else if (!strcmp(str, "all")) {
			for (int i = 1; i < 21; i++)
				flag |= 1 << i;
		}
		else if (!strcmp(str, "empty"))
			flag = 0;
	}
	return 0;
}
